umh.docs.umh.app/content/en/docs/architecture/microservices/grafana-plugins/umh-datasource-v2.md

```diff
@@ -0,0 +1,61 @@
+---
+title: "Umh Datasource V2"
+content_type: grafana-plugin
+description: |
+    This page contains the technical documentation of the umh-datasource-v2 plugin,
+    which allows for easy data extraction from factoryinsight.
+weight: 1000
+---
+
+<!-- overview -->
+
+The plugin, **umh-datasource-v2**, is a Grafana data source plugin that allows you to fetch
+resources from a database and build queries for your dashboard.
+
+<!-- body -->
+
+## {{% heading "howitworks" %}}
+
+1. When creating a new panel, select **umh-datasource-v2** from the Data source drop-down menu. It will then fetch the resources
+   from the database. The loading time may depend on your internet speed.
+
+   ![selectingDatasource](/images/grafana-plugins/grafanaPluginsSelectingV2.png?width=80%)
+
+2. Select the resources in the cascade menu to build your query. **DefaultArea** and **DefaultProductionLine** are placeholders
+   for the future implementation of the new data model.
+
+   ![selectingDatasource](/images/grafana-plugins/grafanaPluginsSelectingWorkCell.png?width=80%)
+
+3. Only the available values for the specified work cell will be fetched from the database. You can then select which data value you want to query.
+
+   ![selectingDatasource](/images/grafana-plugins/grafanaPluginsSelectingValue.png?width=80%)
+
+4. Next you can specify how to transform the data, depending on what value you selected.
+   Below you can find the options to transform the queried data:
+
+   - **Time bucket**: lets you group data in a time bucket
+   - **Aggregates**: common statistical aggregations (maximum, minimum, sum or count)
+   - **Handling missing values**: lets you choose how missing data should be handled
+   - **Keep state integer**. lets you keep the state as an integer instead of converting it into its corresponding string
+   - **Include running processes**: includes the running processes
+   - **Keep states**: keeps the state as an integer or string
```

Comment:
What is the differneceo f this to keep state integer? When do these options appear and what are they for?

Reply:
There is no documentation for "Keep state integer", "Include running processes" and "Keep states"

Reply:
Where do they show then? If you are unsure, you can ask @DaruZero .

Reply:
- **Keep state integer** is available only for standard tags. If enabled, instead of showing the string representation of the state, it shows its integer value. e.g. instead of "No shifts", returns 160000
- **Keep state** works the same way, but only for availability tables, like availabilityHistogram and availabilityTotal